src/main/java/xbot/common/injection/DaggerRobotModule.java

```diff
@@ -0,0 +1,48 @@
+package xbot.common.injection;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import dagger.Binds;
+import dagger.Module;
+import xbot.common.command.RealSmartDashboardCommandPutter;
+import xbot.common.command.SmartDashboardCommandPutter;
+import xbot.common.controls.sensors.XTimerImpl;
+import xbot.common.controls.sensors.wpi_adapters.TimerWpiAdapter;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.logging.SilentRobotAssertionManager;
+import xbot.common.properties.ITableProxy;
+import xbot.common.properties.PermanentStorage;
+import xbot.common.properties.PreferenceStorage;
+import xbot.common.properties.SmartDashboardTableWrapper;
+import xbot.common.properties.TableProxy;
+import xbot.common.properties.XPropertyManager;
+
+@Module
+public abstract class DaggerRobotModule {
```

Comment:
I was having a hard time wrapping my head around how these Modules relate to the `BaseComponent` class

Reply:
Now that I've finished the remainder of the migration, hopefully the usage is clearer.